modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/resource/v1_0/OrderItemResourceImpl.java

```diff
@@ -321,34 +321,48 @@ public OrderItem putOrderItem(Long id, OrderItem orderItem)
 				_commerceOrderItemService.updateCommerceOrderItemPrices(
 					commerceOrderItem.getCommerceOrderItemId(),
 					(BigDecimal)GetterUtil.getNumber(
-						orderItem.getDiscountAmount()),
+						orderItem.getDiscountAmount(), BigDecimal.ZERO),
```

Comment:
Ehi @amosfong, thanks for you review I've had a look at your changes here https://github.com/liferay-commerce/liferay-portal/pull/2585/ I think you catched a bug, but I think the fix is not right. I believe that those changes are fine for a PATCH method not for the PUT, and I think the PATCH is already implemented for this endpoint, so I've made this, which will make the proper update for the PUT method.

Please let me know if this works for you, thanks!

Reply:
ok yea, this makes more sense, thanks!